lib/project_types/script/layers/infrastructure/api_clients.rb

```diff
@@ -0,0 +1,92 @@
+module Script
+  module Layers
+    module Infrastructure
+      class ApiClients
+        def self.default_client(ctx, api_key)
+          if ENV["BYPASS_PARTNERS_PROXY"]
+            PartnersProxyBypassAPI.new(ctx, api_key)
+          else
+            PartnersProxyAPI.new(ctx, api_key)
+          end
+        end
+
+        class PartnersProxyBypassAPI < ShopifyCli::API
```

Comment:
`ScriptServiceAPI` seems pretty straightforward to me, since it directly hits SS and nothing else. Though I also get using `PartnersProxyBypassAPI` since you're right that it encapsulates our abstraction. 

I'd probably avoid `LocalDevelopmentAPI` since `PartnersProxyAPI` can hit local Partners depending on your env vars. 

> One thing I noticed while in this code is that BYPASS_PARTNERS_PROXY doesn't appear to be gated by the environment, which means someone could do this in production, no? It's possible we should also check to make sure we are actually in development on top of the bypass flag.

I think thats fine 🤷‍♂️. Even if they use that flag while using the gem/brew package, it would still hit local SS and would do no harm to the user. Problems would only occur if for some reason they have that env var setup themselves for some other reason (seems unlikely), but they still want to hit prod. I do find it useful using the published CLI for testing locally though. Maybe it needs a prefix `SHOPIFY_CLI` prefix?